lib/egis/aws_client_provider.rb

```diff
@@ -6,6 +6,10 @@
 module Egis
   # @!visibility private
   class AwsClientProvider
+    def initialize(configuration)
+      @configuration = configuration
+    end
```

Comment:
I think it would be better to pass configuration object to the factory method since it is not a singleton anymore.

```ruby
def s3_client(egis_configuration)
   ...
end
```